main/views.py

```diff
@@ -235,3 +260,162 @@ def delete_profile(request):
     except Exception as e:
         print(e)
         return JsonResponse({"status": 500, "message": "Profile deletion unsuccessful"})
+
+@csrf_exempt
+@auth_required
+def create_reimbursement(request):
+    try:
+        user = request.user
+        res = json.loads(request.body)
+        amount = res["amount"]
+        reason = res["reason"]
+        type_of = res["type"]
+        expense_date = res["expense_date"]
+        reimbursement_data = Reimbursement.objects.create(user=user, amount=amount, reason=reason, type=type_of, expense_date=expense_date)
+        return JsonResponse({"status": 201, "data": reimbursement_data.to_dict()})
+    except Exception as e:
+        print(e)
+        return JsonResponse({"status": 500, "message": "Please enter correct details!!!"})
+
+@csrf_exempt
+@auth_required
+def get_reimbursements(request):
+    try:
+        user = request.user
+        reimbersments = Reimbursement.objects.filter(user=user)
+        reim_dict = [reim.to_dict() for reim in reimbersments]
+        return JsonResponse({"status": 200, "message": reim_dict})
+    except Exception as e:
+        print(e)
+        return JsonResponse({"status": 500, "message": "Reimbursement not found"})
+
+@csrf_exempt
+@auth_required
+def edit_reimbursement(request):
+    try:
+        user = request.user
+        res = json.loads(request.body)
+        reimbursement_id = int(res["id"])
+        reimbursement = Reimbursement.objects.filter(id=reimbursement_id, user=request.user).first()
```

Comment:
check if the reimbursment is approved, allow no reimbursments after being approved

Reply:
ok